### PR TITLE
implement list all api and test case

### PR DESCRIPTION
**Pull request by joanne3634**
1. add serialize and deserialize since list all need
2. implement list all api
3. correct the inventory factory type of product id
4. test serialize and deserialize
5. test list all inventory

### CODE REVIEW

File: service/models.py [hunk 1]
```diff
@@ -71,6 +71,32 @@ def save(self):
...
+        Args:
+            data (dict): A dictionary containing the Inventory data
+        """
+        try:
+            self.product_id = data['product_id']
+            self.quantity = data['quantity']
+            self.restock_level = data['restock_level']
+            self.condition = data['condition']
+            self.available = data['available']
+        except TypeError as error:
```
Comment (eatingzhang0611) [inline service/models.py]: Also need to catch KeyError🤔

Comment (joanne3634) [reply service/models.py]: sure! done

**Comment by joanne3634** [comment]
adding testcases for deserialize @eatingzhang0611 @faberald

**Comment by joanne3634** [comment]
@eatingzhang0611 @faberald @SonakshiRaizada  sorry keep changing, just separate serialize and deserialize branch to #50, now this branch is just for list api, it should be easier to review, thx

**Pull request merged**